sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/lang/util/TreeUtil.kt

```diff
@@ -151,7 +153,7 @@ inline fun <reified T : PsiElement> PsiElement.nextSiblingOfType(): T {
   return PsiTreeUtil.getNextSiblingOfType(this, T::class.java)!!
 }
 
-private fun PsiElement.rangesToReplace(): List<Pair<IntRange, String>> {
+private fun PsiElement.rangesToReplace(isAsync: Boolean): List<Pair<IntRange, String>> {
```

Comment:
and this parameter should probably be `alwaysUseIndexedVariables` or something like that

Reply:
ie I want our parameters to clearly state the change they want to happen, vs stating the state they're currently in and having an implementation detail gate its logic on that, as much as possible

and if it truly is logic thats gated on some global state of the compiler, we should be querying it from one of the project structure types or something like that instead of passing it around.